src/main/scala/net/heartsavior/spark/sql/state/StateStoreWriter.scala

```diff
@@ -45,8 +45,7 @@ class StateStoreWriter(
   private val storeConf = new StateStoreConf(session.sessionState.conf)
 
   // A Hadoop Configuration can be about 10 KB, which is pretty big, so broadcast it
-  private val hadoopConfBroadcast = session.sparkContext.broadcast(
-    new SerializableConfiguration(session.sessionState.newHadoopConf()))
+  private val hadoopConfBroadcast = new SerializableConfigurationWrapper(session)
```

Comment:
Would it work if we don't change the entire code but change `session.sessionState.newHadoopConf()` to `session.sparkContext.hadoopConfiguration`? That might be the point where bug hides.

Reply:
I tried that first and it didn't work.

Reply:
Sigh. Thanks for letting me know. Let's add some comment to the wrapper class to avoid missing the needs for the class.